README.md

```diff
@@ -17,85 +19,72 @@ make build install
-
-1. Open, or create, `~/.porter/config.toml`.
-
-1. Add the following lines:
-
-    ```toml
-    default-storage = "kubernetes-storage"
+The plugin also requires that the user or service account that is being used with Kubernetes has `"get","list","create","delete",` and `"patch"` permissions on secrets in the namespace.
 
-    [[storage]]
-    name = "kubernetes-storage"
-    plugin = "kubernetes.storage"
-    ```
+The [Porter Operator](https://github.com/getporter/operator) is the primary use case
+for running in Kubernetes which configures the necessary service accounts via 
+it's `configureNamespace` custom action.
 
-* If the plugin is being used outside of a Kubernetes cluster then add the following lines to specify the namespace to be used to store data:
+```
+porter invoke porterops --action configureNamespace --param namespace=quickstart -c porterops
+```
 
-    ```toml
-    [storage.config]
-    namespace = "<namespace name>"
-    ```
 
 ### Secrets
 
-The `Kubernetes.secrets` plugin enables resolution of credential or parameter values as secrets in Kubernetes.
+The `kubernetes.secrets` plugin enables resolution of credential or parameter values as secrets in Kubernetes via the Porter Operator.
 
-1. Open, or create, `~/.porter/config.toml`
+1. Create, `./porter-k8s-config.yaml`
 1. Add the following lines1:
 
-    ```toml
-    default-secrets = "kubernetes-secrets"
+    ```yaml
+    default-secrets: "kubernetes-secrets"
+    secrets:
+    - name: "kubernetes-secrets"
+      plugin: "kubernetes.secrets"
+    ```
+1. Provide the Porter config to the `configureNamespace` operator bundle action
 
-    [[secrets]]
-    name = "kubernetes-secrets"
-    plugin = "kubernetes.secret"
+    ```
+    porter invoke operator --action=configureNamespace --param namespace=<namespace name> --param porterConfig=porter-k8s-config.toml -c kind -n=operator
     ```
 
 * If the plugin is being used outside of a Kubernetes cluster then add the following lines to specify the namespace to be used to store data:
 
-    ```toml
-    [secrets.config]
-    namespace = "<namespace name>"
+    ```yaml
+    default-secrets: "kubernetes-secrets"
+    secrets:
+      - name: "kubernetes-secrets"
+        plugin: "kubernetes.secrets"
+        config:
+          namespace: "<namespace name>"
     ```
 
-### Storage and Secrets combined
-
-When both storage and secrets are configured, be sure to place the `default-*` stanzas
-at the top of the file, like so:
-
-  ```toml
-  default-secrets = "kubernetes-secrets"
-  default-storage = "kubernetes-storage"
-
-  [[secrets]]
-  name = "kubernetes-secrets"
-  plugin = "kubernetes.secret"
-
-  [[storage]]
-  name = "kubernetes-storage"
-  plugin = "kubernetes.storage"
-  ```
-
-If runing outside of Kubernetes then also include the namespace configuration
-  
-  ```toml
-  [secrets.config]
-  namespace = "<namespace name>"
+In both cases the Kubernetes secret must be created with a `credential` key
```

Comment:
This is what I never understood about the plugin before. Thanks for spelling it out.

But now I'm wondering if we should change the key to something that makes sense for both parameter and credential sets. Possible ideas: `porter-secret`, `value`, `SECRET_NAME` (i.e. if the secret value in the credential set is sadpanda, then the k8s secret is named sadpanda and the key would be sadpanda.

What do you think?

Reply:
Value sounds good to me!